### PR TITLE
Adds test to http reporter.

### PR DESCRIPTION
**Pull request by jcchavezs**
This PR adds a test for the http reporter.

Ping @basvanbeek

### CODE REVIEW

File: model/span.go [hunk 1]
```diff
@@ -30,20 +30,26 @@ type SpanModel struct {
 	Shared         bool              `json:"shared"`
 	LocalEndpoint  *Endpoint         `json:"localEndpoint,omitempty"`
 	RemoteEndpoint *Endpoint         `json:"remoteEndpoint,omitempty"`
-	Annotations    []Annotation      `json:"annotations"`
-	Tags           map[string]string `json:"tags"`
+	Annotations    []Annotation      `json:"annotations,omitempty"`
+	Tags           map[string]string `json:"tags,omitempty"`
```
Comment (jcchavezs) [inline model/span.go]: We dont' want to send `annotations:null` nor `tags:null` in the payload.

File: model/span.go [hunk 2]
```diff
@@ -30,20 +30,26 @@ type SpanModel struct {
...
 }
 
 // MarshalJSON exports our Model into the correct format for the Zipkin V2 API.
 func (s SpanModel) MarshalJSON() ([]byte, error) {
 	type Alias SpanModel
+
+	var timestamp uint64
+	if !s.Timestamp.IsZero() {
```
Comment (jcchavezs) [inline model/span.go]: The reason for this change is because on no timestamp we were marshalling it as `-6795364578871345`.

File: model/span.go (hunk 1 above)
Comment (basvanbeek) [reply model/span.go]: Agreed... this has already been handled with in #23

File: model/span.go (hunk 2 above)
Comment (basvanbeek) [reply model/span.go]: good call... I've found some other issues in model as well which made sense to handle in #23 with unit tests added for checking these edge cases.

**Comment by basvanbeek** [comment]
@jcchavezs LGTM, I'm going to merge #23 and can you rebase this PR afterwards?

**Comment by jcchavezs** [comment]
@basvanbeek rebased.
